services/queue/src/api.js

```diff
@@ -1995,6 +1998,47 @@ builder.declare({
   });
 });
 
+/** Pending and claimed counts for a queue */
+builder.declare({
+  method: 'get',
+  route: '/task-queues/:taskQueueId(*)/counts',
+  name: 'taskQueueCounts',
+  scopes: {
+    AllOf: [
+      'queue:pending-count:<taskQueueId>',
+      'queue:claimed-count:<taskQueueId>',
+    ],
+  },
+  stability: APIBuilder.stability.stable,
+  category: 'Worker Metadata',
+  output: 'task-queue-counts-response.yml',
+  title: 'Get Number of Pending and Claimed Tasks',
+  description: [
+    'Get an approximate number of pending and claimed tasks for the given `taskQueueId`.',
+    '',
+    'As task states may change rapidly, this number may not represent the exact',
+    'number of pending and claimed tasks, but a very good approximation.',
+  ].join('\n'),
+}, async function(req, res) {
+  const taskQueueId = req.params.taskQueueId;
+  const { provisionerId, workerType } = splitTaskQueueId(taskQueueId);
+
+  // Get number of pending message
+  let [pendingCount, claimedCount] = await Promise.all([
```

Comment:
Prefer `const` here I think

```suggestion
  const [pendingCount, claimedCount] = await Promise.all([
```